### PR TITLE
Distributed transpose

### PR DESCRIPTION
**Pull request by rahulghangas**

### CODE REVIEW

**Comment by ankingcodes** [comment]
@ben-albrecht wrong label

File: modules/packages/LinearAlgebra.chpl [hunk 1]
```diff
@@ -503,8 +503,44 @@ proc transpose(A: [?Dom] ?eltType) where isDenseMatrix(A) {
   }
 }
 
+pragma "no doc"
+proc tranpose(A: [?Dom] ?eltType, keepDistribution = true)
```
Comment (e-kayrakli) [inline modules/packages/LinearAlgebra.chpl:507]: Maybe `transposeDistribution` or `transposeDist` is a better name for this flag? Obivously, the default would be `false` in that case.

That argument can also be made `param`, but I do not have a strong preference one way or another.

File: modules/packages/LinearAlgebra.chpl [hunk 2]
```diff
@@ -503,8 +503,44 @@ proc transpose(A: [?Dom] ?eltType) where isDenseMatrix(A) {
...
+  const transposedDom = {Dom.dim(1), Dom.dim(0)};
+  const targetLocales = A.targetLocales();
+
+  if keepDistribution {
+    var distDom = transposedDom dmapped Block(boundingBox=transposedDom, 
+                                              targetLocales=targetLocales);
+    var transposedMat : [distDom] eltType;
+
+    coforall loc in targetLocales {
+      var localDomain = A.localSubdomain();
```
Comment (e-kayrakli) [inline modules/packages/LinearAlgebra.chpl:520]: Both out of principle and consistency, I'd make this a `const`.

FYI: we are working on finding ways of exploiting constness of domains to do some performance optimizations. I don't think it matters for this case, but in general there might be some benefits of using const domains in the future if that makes sense for the specific application.

File: modules/packages/LinearAlgebra.chpl [hunk 3]
```diff
@@ -503,8 +503,44 @@ proc transpose(A: [?Dom] ?eltType) where isDenseMatrix(A) {
   }
 }
 
+pragma "no doc"
+proc tranpose(A: [?Dom] ?eltType, keepDistribution = true) 
+  where isDistributed(A) 
+{
+  // Add case for rank 1
```
Comment (e-kayrakli) [inline modules/packages/LinearAlgebra.chpl:510]: I believe this is a TODO? Are you hoping to do that in this PR, if not maybe add an error for now.

**Comment by e-kayrakli** [comment]
Hi @rahulghangas -- will you be able to make some progress on the open discussions on this PR soon? If not, I'd encourage closing the PR and reopen once you have the time to work on it.

**Comment by e-kayrakli** [comment]
Based on silence, I am closing this PR.

@rahulghangas -- Please feel free to reopen when you have the time to work on this.

**Pull request closed without merging**